### PR TITLE
Fix Who We Are section on mobile

### PR DESCRIPTION
**Pull request by Satoshi-Sh**
|                                    Web Dev Path                                    |
| :--------------------------------------------------------------------------------: |
| [**189**](https://github.com/MarianaSouza/web-dev-path/issues/189) |

#### Have you updated the CHANGELOG.md file? If not, please do it.
I did.
#### What is this change?
Lowered the threshold to trigger the animation of the developer profile section. 
#### Were there any complications while making this change?
No.
#### How did you verify this change?
Checked with the browser mobile simulator. 
#### When should this be merged?
After the review.
#### Note:
It would be great if you can check the section on your mobile.

### CODE REVIEW

**Comment by mariana-caldas** [comment]
Hey, @Satoshi-Sh !

I appreciate the effort in addressing the visibility issue of the `RevealContentContainer` by adjusting the threshold to `0.08`. While this does solve the current problem, I'd like to propose a more flexible solution that can cater to sections of varying heights.

**The Problem:**
The IntersectionObserver's threshold represents the percentage of the target element that is visible within the root's intersection rectangle. A static threshold might not always be optimal as sections grow or shrink.

**Proposed Solution: Multiple Thresholds**
Instead of a single threshold, we can use an array of thresholds, ensuring the callback is triggered at various visibility points.

**Implementation:**

_Hook useIntersect:_

```
import { useState, useEffect, useRef } from 'react';

export const useIntersect = ({ root = null, rootMargin }) => {
  const [entry, setEntry] = useState({});
  const [node, setNode] = useState(null);

  let observer = useRef(null);

  // Create an array of thresholds from 0.01 to 0.99
  const threshold = Array.from({ length: 100 }, (_, index) => (index + 1) * 0.01);

  useEffect(() => {
    observer = new window.IntersectionObserver(
      ([entry]) => {
        setEntry(entry);
      },
      { root, rootMargin, threshold }
    );

    if (node) {
      observer.observe(node);
    }
    return () => {
      observer.disconnect();
      if (node) {
        observer.unobserve(node);
      }
    };
  }, [node, root, rootMargin]);

  return [setNode, entry];
};
```

_Component RevealContentContainer_

The `RevealContentContainer` component receives the hook as an empty object instead of having a hardcoded threshold.

`const [ref, entry] = useIntersect({});`

Please let me know wif that makes sense, Satoshi.

**Comment by Satoshi-Sh** [comment]
Thanks for your help, @mariana-caldas. I changed the code as you instructed. Please have a look when you have time. :+1:

File: hooks/useIntersect.js [hunk 1]
```diff
@@ -23,7 +27,7 @@ export const useIntersect = ({ root = null, rootMargin, threshold = 0 }) => {
         observer.unobserve(node);
       }
     };
-  }, [node, root, rootMargin, threshold]);
+  }, [node, root, rootMargin]);
```
Comment (mariana-caldas) [inline hooks/useIntersect.js:30]: Can you please add back the threshold property here, @Satoshi-Sh ? My example did not have it but it should be included in the dependency array of the `useEffect` hook to ensure that the effect re-runs if the threshold changes. Thanks!

**Review by mariana-caldas — CHANGES_REQUESTED** [review]

**Review by mariana-caldas — APPROVED** [review]
LGTM! Thanks, @Satoshi-Sh !

**Review by LuSilvaDeveloper — APPROVED** [review]
Everything is working properly here on mobile view.

**Review by sergiobtos — APPROVED** [review]
LGTM. For me, it's working. Interesting code. I had never used this way before. Nice.

**Pull request merged**